src/index.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  getInput,
+  getMultilineInput,
+  info,
+  setFailed,
+  endGroup,
+  startGroup,
+} from "@actions/core";
+import { spawnSync } from "node:child_process";
+import { existsSync } from "node:fs";
+import * as path from "node:path";
+import * as util from "util";
+const exec = util.promisify(require("child_process").exec);
+
+const DEFAULT_WRANGLER_VERSION = "^3.0.0";
+
+/**
+ * A configuration object that contains all the inputs & immutable state for the action.
+ */
+const config = {
+  WRANGLER_VERSION:
+    getInput("wranglerVersion") !== ""
+      ? getInput("wranglerVersion")
+      : DEFAULT_WRANGLER_VERSION,
+  secrets: getMultilineInput("secrets"),
+  workingDirectory: checkWorkingDirectory(getInput("workingDirectory")),
+  CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+  CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+  ENVIRONMENT: getInput("environment"),
+  VARS: getMultilineInput("vars"),
+  COMMANDS: getMultilineInput("commands"),
+} as const;
+
+async function main() {
+  installWrangler();
+  authenticationSetup();
+  await execCommands(getMultilineInput("preCommands"));
+  await uploadSecrets();
+  await wranglerCommands();
+  await execCommands(getMultilineInput("postCommands"));
+}
+
+function checkWorkingDirectory(workingDirectory = "") {
+  try {
+    const normalizedPath = path.normalize(workingDirectory);
+    if (existsSync(normalizedPath)) {
+      return normalizedPath;
+    } else {
+      setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+    }
+  } catch (error) {
+    setFailed(
+      `🚨 While checking/creating directory ${workingDirectory} received ${error}`
+    );
+  }
+}
+
+function installWrangler() {
+  if (config["WRANGLER_VERSION"].startsWith("1")) {
+    setFailed(`🚨 Wrangler v1 is no longer supported by this action. Please use major versions 2 or 3`);
+  }
+  startGroup("📥 Installing Wrangler");
+  const command = `npx install wrangler@${config["WRANGLER_VERSION"]}`;
+  info(`Running Command: ${command}`);
+  spawnSync(command, { cwd: config["workingDirectory"], env: process.env });
+  endGroup();
+}
+
+function authenticationSetup() {
+  startGroup("🔐 Authenticating with Cloudflare");
+  try {
+    process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_ACCOUNT_ID"];
+    process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_API_TOKEN"];
+    info(`Authentication process initiated with - API Token`);
+  } catch (error) {
+    setFailed(
+      `Authentication details were not found. Please input an 'apiToken' to the action.`
+    );
+  }
+  endGroup();
+}
+
+async function execCommands(commands: string[]) {
+  if (!commands.length) {
+    return;
+  }
+
+  startGroup("🚀 Pre/Post Commands Group");
+
+  const arrPromises = commands.map(async (command) => {
+    const npxCMD = command.startsWith("wrangler") ? `npx ${command}` : command;
```

Comment:
Further down you have a different `npx` command for Windows—is that needed here?